sonar-scala-plugin/src/main/java/org/sonarsource/scala/plugin/ScoverageSensor.java

```diff
@@ -0,0 +1,182 @@
+
+  private static final Logger LOG = Loggers.get(ScoverageSensor.class);
+
+  private static final int MAX_LOGGED_FILE_NAMES = 20;
+
+  @Override
+  public void describe(SensorDescriptor descriptor) {
+    descriptor
+        .onlyWhenConfiguration(conf -> conf.hasKey(COVERAGE_REPORT_PATHS_KEY))
+        .name("Scoverage sensor for Scala coverage");
+  }
+
+  @Override
+  public void execute(SensorContext context) {
+    List<File> reportFiles = ExternalReportProvider.getReportFiles(context, COVERAGE_REPORT_PATHS_KEY);
+
+    if (reportFiles.isEmpty()) {
+      //Return if we don't have any files
+      return;
+    }
+
+    Set<String> unresolvedInputFile = new HashSet<>();
+
+    for (File f : reportFiles) {
+      if(f.getName().endsWith(XML_REPORT_FILENAME)) {
+        //Read the scoverage.xml file
+        try(InputStream in = new FileInputStream(f)){
+          readReportFile(in, context, unresolvedInputFile);
+        } catch (IOException | XMLStreamException e ) {
+          LOG.error("File '{}' can't be read. " + e.getMessage(), f, e);
+        }
+      }
+    }
+
+    logUnresolvedInputFiles(unresolvedInputFile);
+  }
+
+  private static void readReportFile(InputStream in, SensorContext context, Set<String> unresolvedInputFile) throws XMLStreamException {
+    Map<String, Set<Integer>> linesHitPerFiles = new HashMap<>();
+
+    XMLEventReader reader = XMLInputFactory.newInstance().createXMLEventReader(in);
+    while (reader.hasNext()) {
+      XMLEvent event = reader.nextEvent();
+      if (event.isStartElement() && STATEMENT_ELEMENT.equals(event.asStartElement().getName())) {
+        //We have a statement, read the attributes of the statement
+        parseStatementAttributes(linesHitPerFiles, event.asStartElement());
+      }
+    }
+    //Once we have parsed all this report, add the statements to the context
+    addLineHitToContext(linesHitPerFiles, context, unresolvedInputFile);
+  }
+
+
+  private static void parseStatementAttributes(Map<String, Set<Integer>> linesHitPerFiles, StartElement currentEvent){
+    Integer line = null;
+    Integer invocationCount = null;
+    String source = null;
+    Iterator<Attribute> attributes = currentEvent.getAttributes();
+    while (attributes.hasNext()) {
+      Attribute attribute = attributes.next();
+      if (attribute.getName().equals(INVOCATION_COUNT_ATTRIBUTE)) {
+        invocationCount = Integer.valueOf(attribute.getValue());
+      }
+      if (attribute.getName().equals(LINE_ATTRIBUTE)) {
+        line = Integer.valueOf(attribute.getValue());
+      }
+      if (attribute.getName().equals(SOURCE_ATTRIBUTE)) {
+        source = attribute.getValue();
+      }
+    }
+    if(line != null && invocationCount != null && source != null) {
+      addStatementToMap(linesHitPerFiles, source, line, invocationCount);
+    } else {
+      LOG.warn("Some attributes of statement at line {} of scoverage report are not present.", currentEvent.getLocation().getLineNumber());
```

Comment:
My fear is that: if we import a coverage report with a format that we don't support, if the project has 1M executable line of code, we will have 1M of warning in the log :grimacing: